Use MIMA

cstamas opened the pull request:
Solves #217

A rough draft how could it look like, but I see a big influence of shrinkwrap, hence, IMHO this could be radically simplified.

Also, unsure is "LocalRepository" layrri specific (I guess it is), but just FYI: new feature of resolver is "chained local repository", so it is able to use multiple maven local repositories as well, might be worth to factor it in.

cstamas committed:
Use MIMA

Solves #217

cstamas committed:
Slight fixes

cstamas committed:
Added TODO

aalmiray commented:
FWIW local repository is just a dir with JARs with or without POMs. We need this option to resolve JARs when network is not available. There may not be POMs nor maven-metadata.xml and that's fine.

cstamas commented:
Hm, it seems GH env gets detected and as those things are at "standard places" MIMA tries (and fails) to decrypt them?
```
Failed to decrypt password for server github: org.sonatype.plexus.components.sec.dispatcher.SecDispatcherException: java.io.FileNotFoundException: /home/runner/.m2/settings-security.xml (No such file or directory) @ server: github
```

My assumption (unverified):
* settings.xml discovered by MIMA has `github` credentials encrypted
* there is no `settings-security.xml` present, hence MIMA fails
* it is telling that in module layrry-integrationtest-runner I see no settings.xml at all (hence I believe this is GH Maven env settings that is being discovered)

@aalmiray can you confirm?

EDIT: https://github.com/maveniverse/mima/issues/2

cstamas committed:
Use MIMA 2.1.0

That has several crypto related bugs fixed:
https://github.com/maveniverse/mima/releases/tag/release-2.1.0
